backend/test/animal.e2e-spec.ts

```diff
@@ -0,0 +1,108 @@
+                    AnimalModule
+            ],
+		}).compile();
+
+		app = moduleFixture.createNestApplication();
+		await app.init();
+
+        animalController = moduleFixture.get<AnimalController>(AnimalController);
+            // seed db with animals for each testcase
+        for (const animal in testAnimals) {
+            await animalController.create({name: testAnimals[animal]})
+        }
+	});
+
+    afterAll(async () => {
+        app.close();
+    })
+    
+    describe("/test  (GET)", () => {
+        it("it should retrun an array of all users", () => {
+            return request(app.getHttpServer())
+                .get("/test")
+                .expect(HttpStatus.OK)
+        })
+    })
+
+    describe("/test/3  (GET)", () => {
+        it("it should retrun an array of all users", () => {
+            return request(app.getHttpServer())
+                .get("/test/3")
+                .expect(HttpStatus.OK)
+                .expect((response: request.Response) => {
+                    const {
+                        id,
+                        name
+                    } = response.body;
+                    expect(typeof id).toBe("number"),
+                    expect(name).toEqual(testAnimals[2]);
+                })
+        })
+    })
+    
+    describe("/test  (POST)", () => {
+        it("it should return a the created user data", () => {
+            return request(app.getHttpServer())
+                .post("/test")
+                .set('Accept', 'application/json')
+                .send(mockAnimal)
+                .expect(HttpStatus.CREATED)
+                .expect((response: request.Response) => {
+                    const {
+                        id,
+                        name
+                    } = response.body;
+                    expect(typeof id).toBe("number"),
+                    expect(name).toEqual(mockAnimal.name);
+                })
+            })
+        })
+        
+        describe("/test/1  (delete)", () => {
+            it("it should delete a user", () => {
+                return request(app.getHttpServer())
+                .delete("/test/1")
+                .expect(HttpStatus.OK)
+            })
+        })
+        
+        describe("/test/2  (patch)", () => {
+            it("update animal with id 2", () => {
+                const updateData: UpdateAnimalDto = {name: "Nagapie"}
+                return request(app.getHttpServer())
+                    .patch("/test/2")
+                    .set('Accept', 'application/json')
+                    .send(updateData)
+                    .expect(HttpStatus.OK)
+                    .expect((response: request.Response) => {
+                        const {
+                            affected
+                        } = response.body;
+                        expect(affected).toEqual(1);
+                    })
+                })
+            })
+});
```

Comment:
so far this seems to be only testing behaviour that is going well.
should we add tests for things that don't go well?

GET/DELETE/PATCH for ID's that don't exist
POST with wrong/empty body

something like this:
```ts
describe('/test/6 (GET)', () => {
    it('should return a 404', () => {
        return request(app.getHttpServer())
            .get('/test/6')
            .expect(HttpStatus.NOT_FOUND);
    });
});
```

now is a failing test, since the service just returns NULL, which is what is also returned to the enduser.. So we get a 200 OK response with an empty body.. 
Not sure if that is the behaviour we want.. Also not sure if we want to respond with 404.. but still